kpack-image-builder/controllers/builderinfo_controller.go

```diff
@@ -110,7 +110,7 @@ func (r *BuilderInfoReconciler) ReconcileResource(ctx context.Context, info *kor
 			Type:    ReadyConditionType,
 			Status:  metav1.ConditionFalse,
 			Reason:  "ClusterBuilderNotReady",
-			Message: fmt.Sprintf("ClusterBuilder %q is not ready", r.clusterBuilderName),
+			Message: fmt.Sprintf("ClusterBuilder %q is not ready: %s", r.clusterBuilderName, clusterBuilderReadyCondition.Message),
```

Comment:
Since this statement is in an else clause with clusterBuilderReadyCondition != nil and clusterBuilderReadyCondition.Status == corev1.ConditionTrue, it's possible that clusterBuilderReadyCondition could be nil here right? So we may have to guard against that when using clusterBuilderReadyCondition.Message.

Reply:
Good catch. I've added a test for this and updated the code to detect whether there is a ready condition on the ClusterBuilder.